org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingSuperAfterStatementsTest.java

```diff
@@ -0,0 +1,209 @@
+        argvParameter.setType(ast.newArrayType(ast.newSimpleType(ast.newSimpleName("String"))));
+        argvParameter.setName(ast.newSimpleName("argv"));
+        mainMethod.parameters().add(argvParameter);
+
+        Block mainMethodBody = ast.newBlock();
+        ClassInstanceCreation instanceCreation = ast.newClassInstanceCreation();
+        instanceCreation.setType(ast.newSimpleType(ast.newSimpleName("X")));
+        instanceCreation.arguments().add(ast.newNumberLiteral("0"));
+
+        mainMethodBody.statements().add(ast.newExpressionStatement(instanceCreation));
+        mainMethod.setBody(mainMethodBody);
+
+        // Add main method to class
+        classDeclaration.bodyDeclarations().add(mainMethod);
+
+        // Add class to compilation unit
+        compilationUnit.types().add(classDeclaration);
+
+		StringBuilder buf = new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("class X {\n");
+		buf.append("  X(  int i){\n");
+		buf.append("    if (i < 0) {\n");
+		buf.append("      i++;\n");
+		buf.append("    }\n");
+		buf.append("    super();\n");
+		buf.append("  }\n");
+		buf.append("  public void main(  String[] argv){\n");
+		buf.append("    new X(0);\n");
+		buf.append("  }\n");
+		buf.append("}\n");
+
+		assertEqualString(compilationUnit.toString(), buf.toString());
+	}
+
+	public void test002() throws Exception {
+		IPackageFragment pack1= this.sourceFolder.createPackageFragment("test1", false, null);
+		StringBuilder buf = new StringBuilder();
+		buf= new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("class X {\n");
+		buf.append("  int i;\n");
+		buf.append("  X() {\n");
+		buf.append("    this.i++;\n");
+		buf.append("    this.hashCode();\n");
+		buf.append("  }\n");
+		buf.append("}\n");
+
+		ICompilationUnit cu= pack1.createCompilationUnit("X.java", buf.toString(), false, null);
+        CompilationUnit astRoot= createAST(this.apiLevel, cu);
+        ASTRewrite rewrite= ASTRewrite.create(astRoot.getAST());
+        AST ast= astRoot.getAST();
+
+        assertTrue("Parse errors", (astRoot.getFlags() & ASTNode.MALFORMED) == 0);
+		TypeDeclaration type= findTypeDeclaration(astRoot, "X");
+		MethodDeclaration methodDecl= findMethodDeclaration(type, "X");
+		Block block= methodDecl.getBody();
+		List<Statement> blockStatements = block.statements();
+		assertEquals("Incorrect number of statements", 2, blockStatements.size());
+		{
+			//replace the hashCode() with super()
+			ExpressionStatement	hashCodeStatement = (ExpressionStatement) blockStatements.get(1);
+			SuperConstructorInvocation superInvocation = ast.newSuperConstructorInvocation();
+
+			rewrite.replace(hashCodeStatement, superInvocation, null);
+			System.out.println();
+		}
+		String preview = evaluateRewrite(cu, rewrite);
+		buf= new StringBuilder();
+		buf.append("package test1;\n");
+		buf.append("class X {\n");
+		buf.append("  int i;\n");
+		buf.append("  X() {\n");
+		buf.append("    this.i++;\n");
```

Comment:
This code is wrong - "this" cannot come before super. Unless this test is intended to be written to test this part (I don't think so), please correct - use the runconform tests from the corresponding compiler tests.